pkg/sources/git/git.go

```diff
@@ -522,7 +522,26 @@ func (s *Git) ScanCommits(ctx context.Context, repo *git.Repository, path string
 		repoCtx = context.WithValue(ctx, "repo", path)
 	}
 
-	diffChan, err := s.parser.RepoPath(repoCtx, path, scanOptions.HeadHash, scanOptions.BaseHash == "", scanOptions.ExcludeGlobs, scanOptions.Bare)
+	logger := repoCtx.Logger()
+	var (
+		logValues []any
+
+		baseHashSet bool
+		maxDepthSet bool
```

Comment:
How much time does this save?

Reply:
Not sure but found this a bit easier to read, also felt weird to do the same comparison (regardless of how fast) over and over again only to get the same result.